src/dataframe/dataframe.jl

```diff
@@ -966,9 +966,18 @@ function Base.delete!(df::DataFrame, inds)
     if !isempty(inds) && size(df, 2) == 0
         throw(BoundsError(df, (inds, :)))
     end
+    
     # we require ind to be stored and unique like in Base
     # otherwise an error will be thrown and the data frame will get corrupted
-    foreach(col -> deleteat!(col, inds), _columns(df))
+    n = nrow(df)
+    for col in _columns(df)
+        if length(col) != n
+            throw(ArgumentError("Unequal lengths of columns detected. Most likely the " *
+                                "same column was stored in the data frame several times. 
+                                 The source data frame is now corrupted."))
```

Comment:
Let's continue the user-friendliness contest! :-)
```suggestion
    for i in 1:ncol(df)
        col = _columns(df)[i]
        if length(col) != n
            throw(ArgumentError("Unequal lengths of columns detected at column :$(_names(df)[i]). " *
                                "Most likely the same column was stored in the data frame " *
                                "several times. The source data frame is now corrupted."))
```
The next step is to check which other column is `===` to the current one, and print that too. Oh, and the next one is to skip this column but continue resizing other columns so that the result is *not* corrupt.

Reply:
OK - no rush to merge it. I will make these improvements

Reply:
I have added it :).